src/lndk_offers.rs

```diff
@@ -253,8 +253,11 @@ impl OfferHandler {
 
     /// create_reply_path creates a blinded path to provide to the offer maker when requesting an
     /// invoice so they know where to send the invoice back to. We try to find a peer that we're
-    /// connected to with onion messaging support that we can use to form a blinded path,
-    /// otherwise we creae a blinded path directly to ourselves.
+    /// connected with the necessary requirements to form a blinded path. The peer needs two things:
+    /// 1) Onion messaging support.
+    /// 2) To be an advertised node with at least one public channel.
```

Comment:
Why a node have to have a public channel? Is it because without a channel, node_announcement will never propagate?

Reply:
> Is it because without a public channel, the node_announcement will never propagate and we never get node info?
That's my understanding but I could be wrong there... Currently most of the time the offer node we're communicating with will need to connect directly to the introduction node we provide so we need to make sure the node is in the channel graph.

Reply:
Yes, the issue is that the [spec](https://github.com/lightning/bolts/blob/master/07-routing-gossip.md#requirements-2) says to ignore any `node_announcement` for a node that we a have not seen in a `channel_announcment` to avoid DoS attacks. That's why it needs to have an announced channel, otherwise it won't be in the channel graph.

Reply:
Thanks for the confirmation @dunxen!